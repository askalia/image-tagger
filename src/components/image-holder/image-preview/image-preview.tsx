import { FC, SyntheticEvent } from "react";
import { ImageNotFound } from "../image-not-found/image-not-found";

const ALT_IMAGE_TO_TAG = "this pic is the one to be tagged";

interface IImagePreviewProps {
  imageUrl: string;
  isImageUrlNotFound: boolean;
  coverImageUrlNotFound?: string;
  onImageError: (event: SyntheticEvent) => void;
}

export const ImagePreview: FC<IImagePreviewProps> = ({
  imageUrl,
  isImageUrlNotFound,
  coverImageUrlNotFound,
  onImageError,
}) =>
  isImageUrlNotFound ? (
    <ImageNotFound coverImageUrlNotFound={coverImageUrlNotFound} />
  ) : (
    <img src={imageUrl} alt={ALT_IMAGE_TO_TAG} onError={onImageError} />
  );
