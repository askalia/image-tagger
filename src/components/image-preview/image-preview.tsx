import { FC, SyntheticEvent, useEffect, useState } from "react";
import { ImageNotFound } from "../image-not-found/image-not-found";
import "./image-preview.scss";

const ALT_IMAGE_TO_TAG = "this pic is the one to be tagged";

interface IImagePreviewProps {
  imageUrl: string;
  coverImageUrlNotFound?: string;
  onImageError: (event: SyntheticEvent) => void;
  onClickOverImage: (event: MouseEvent) => void;
}

export const ImagePreview: FC<IImagePreviewProps> = ({
  imageUrl,
  coverImageUrlNotFound,
  onImageError,
  onClickOverImage,
}) => {
  const [isImageUrlNotFound, setIsImageUrlNotFound] = useState<boolean>(false);
  const onError = (error: SyntheticEvent) => {
    console.log("get there");
    setIsImageUrlNotFound(true);

    onImageError(error);
  };

  useEffect(() => {
    setIsImageUrlNotFound(false);
  }, [imageUrl]);
  

  return (
  <div>
    {isImageUrlNotFound === true ? (
      <ImageNotFound
        
        coverImageUrlNotFound={coverImageUrlNotFound}
      />
    ) : (
      <img
        data-cy="image-preview"
        src={imageUrl}
        alt={ALT_IMAGE_TO_TAG}
        onError={onError}
        onLoad={() => console.log("img loaded")}
        onClick={(event: unknown) => onClickOverImage(event as MouseEvent)}
      />
    )}
  </div>)
};
