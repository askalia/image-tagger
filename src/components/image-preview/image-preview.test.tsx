/*import * as React from "react";
import { shallow, mount, render, configure } from "enzyme";

import { ImagePreview } from "./image-preview";
import "@testing-library/jest-dom";

import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

configure({ adapter: new Adapter() });

describe("Animal", () => {
  it("should image src tbe this injected as a props", () => {
    const IMAGE_URL =
      "https://static.hitek.fr/img/actualite/ill_m/944555337/illuspiderman3.jpg";
    const wrapper = render(
      <ImagePreview
        imageUrl={IMAGE_URL}
        onClickOverImage={jest.fn()}
        onImageError={jest.fn()}
      />
    );
    
    //const img = wrapper.find("img");
    expect(wrapper).toBeVisible()
  });
});
*/
import React from "react";
import { shallow, mount, render, configure } from "enzyme";
//import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

//configure({ adapter: new Adapter() });

import { ImagePreview } from "./image-preview";

describe("Hello, Enzyme!", () => {
  it("renders", () => {
    
    const IMAGE_URL =
      "https://static.hitek.fr/img/actualite/ill_m/944555337/illuspiderman3_.jpg";
    const wrapper = render(
      <ImagePreview
        imageUrl={IMAGE_URL}
        onClickOverImage={jest.fn()}
        onImageError={() => console.log('diantre')}
      />
    );
  
    expect(wrapper.find('img').attr('data-testid')).toEqual('image-not-found')
  });
});
