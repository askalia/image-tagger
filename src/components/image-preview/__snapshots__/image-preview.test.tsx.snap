// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal should image src tbe this injected as a props 1`] = `ShallowWrapper {}`;

exports[`Hello, Enzyme! renders snapshots, too 1`] = `
<div>
  <h1>
    Hello, Enzyme!
  </h1>
</div>
`;
